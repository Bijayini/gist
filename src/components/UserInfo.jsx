import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import theme from '../lib/theme';
import BodyCopy from '../ui-components/BodyCopy';

const UserInfo = ({ userDetails }) => {
  return (
    <Fragment>
      <div className="gist-details">
        <div className="bio">
          <img src={userDetails.image} alt={userDetails.owner} />
        </div>
        <div className="details">
          <BodyCopy
            variant="extra-bold"
            xs="x-large"
            xm="x-large"
            xl="x-large"
            color="textColor"
          >
            {userDetails.owner}
          </BodyCopy>
          <br />
          <BodyCopy variant="regular" xs="1" xm="1" xl="1" color="textColor">
            {userDetails.gitHubUrl}
          </BodyCopy>
        </div>
      </div>
      <style jsx="true">
        {`
          .gist-details {
            margin-top: 20px;
            border: 1px solid ${theme.border};
            display: flex;
            padding: 10px;
            border-radius: 3px;
          }
          .bio {
            width: 100px;
            height: 100px;
            margin-right: 20px;
          }
          .bio > img {
            max-width: 100%;
            border-radius: 5px;
          }
        `}
      </style>
    </Fragment>
  );
};

UserInfo.propTypes = {
  userDetails: PropTypes.shape({
    owner: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    gitHubUrl: PropTypes.string.isRequired,
  }).isRequired,
};

export default UserInfo;
