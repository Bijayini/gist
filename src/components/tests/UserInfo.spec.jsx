import React from 'react';
import { shallow } from 'enzyme';

import UserInfo from '../UserInfo';

describe('<UserInfo />', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {
      userDetails: {
        owner: 'some owner',
        image: 'some image',
        gitHubUrl: 'some url',
      },
    };
    component = shallow(<UserInfo {...props} />);
  });

  it('should match with snapshot', () => {
    expect(component.debug()).toMatchSnapshot();
  });
});
