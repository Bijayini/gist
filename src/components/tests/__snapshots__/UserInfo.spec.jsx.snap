// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo /> should match with snapshot 1`] = `
"<Fragment>
  <div className=\\"jsx-413917400 gist-details\\">
    <div className=\\"jsx-413917400 bio\\">
      <img src=\\"some image\\" alt=\\"some owner\\" className=\\"jsx-413917400\\" />
    </div>
    <div className=\\"jsx-413917400 details\\">
      <BodyCopy variant=\\"extra-bold\\" xs=\\"x-large\\" xm=\\"x-large\\" xl=\\"x-large\\" color=\\"textColor\\" className=\\"\\" uppercase={false} styles={false}>
        some owner
      </BodyCopy>
      <br className=\\"jsx-413917400\\" />
      <BodyCopy variant=\\"regular\\" xs=\\"1\\" xm=\\"1\\" xl=\\"1\\" color=\\"textColor\\" className=\\"\\" uppercase={false} styles={false}>
        some url
      </BodyCopy>
    </div>
  </div>
  <JSXStyle id=\\"1132374988\\" dynamic={{...}}>
    .gist-details.__jsx-style-dynamic-selector{margin-top:20px;border:1px solid #c8c8c8;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding:10px;border-radius:3px;}.bio.__jsx-style-dynamic-selector{width:100px;height:100px;margin-right:20px;}.bio.__jsx-style-dynamic-selector&gt;img.__jsx-style-dynamic-selector{max-width:100%;border-radius:5px;}
  </JSXStyle>
</Fragment>"
`;
